feat: Refactors `Button` component (#52)
vtex-sites/gatsby.store: src/components/ui/Button/Button.stories.mdx

import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs'

import Button from '.'
import Icon from 'src/components/ui/Icon'

<Meta
  argTypes={{
    icon: { control: { disable: true } },
  }}
  component={Button}
  title="Atoms/Button"
/>

export const Template = ({ inverse, ...args }) => {
  return inverse ? (
    <div
      style={{
        padding: '10px',
        backgroundColor: '#001947',
      }}
    >
      <Button inverse {...args}>
        Button
      </Button>
    </div>
  ) : (
    <Button {...args}>Button</Button>
  )
}

<header>

# Button

> A element that allows users to take actions, make choices or trigger an event with a single tap.

Reviewer: ```suggestion
This component allows users to take actions, make choices, or trigger an event with a single tap.
```

| -------------------------------------------- | ------------------------------------------------------------------------------------------------- |
| <code>--fs-button-padding</code>             | <code>calc(var(--fs-spacing-1) - (var(--fs-button-border-width) \* 2)) var(--fs-spacing-3)</code> |
| <code>--fs-button-height</code>              | <code>var(--fs-control-tap-size)</code>                                                           |
| <code>--fs-button-gap</code>                 | <code>var(--fs-spacing-2)</code>                                                                  |
|                                              |                                                                                                   |
| <code>--fs-button-shadow</code>              | <code>none</code>                                                                                 |
| <code>--fs-button-shadow-hover</code>        | <code>none</code>                                                                                 |
|                                              |                                                                                                   |

> | <code>--fs-button-border-radius</code>       | <code>var(--fs-border-radius-default)</code>                                                      |

Reviewer: ```suggestion
| <code>--fs-button-border-radius</code>       | <code>var(--fs-border-radius)</code>                                                      |
```

| <code>--fs-button-border-width</code>        | <code>var(--fs-border-width-thick)</code>                                                         |
| <code>--fs-button-border-color</code>        | <code>transparent</code>                                                                          |
|                                              |                                                                                                   |
| <code>--fs-button-text-size</code>           | <code>var(--fs-text-size-base)</code>                                                             |

> | <code>--fs-button-text-weight</code>         | <code>bold</code>                                                                                 |

Reviewer: ```suggestion
| <code>--fs-button-text-weight</code>         | <code>var(--fs-text-weight-bold)</code>                                                                                 |
```

Reviewer: Can you also update here pls? Thanks.

https://github.com/vtex-sites/gatsby.store/blob/d2eab5cf901a9969332c97565b091774ae2d1912/src/components/ui/Button/button.scss#L21

import { ArgsTable, Canvas, Meta, Story } from '@storybook/addon-docs'

import Button, { ButtonBuy, ButtonLink } from '.'
import Icon from 'src/components/ui/Icon'

<Meta
  argTypes={{
    icon: { control: { disable: true } },
  }}
  component={Button}
  title="Atoms/Button"
/>

export const Template = (args) => <Button {...args}>Button</Button>

<header>

# Button

This component allows users to take actions, make choices or trigger an event with a single tap.

</header>

- [Overview](#overview)
- [Usage](#usage)
- [Nested Elements](#nested-elements)
- [Hierarchy](#hierarchy)
- [Variants](#variants)

> - [Related Components](#related-components)

Reviewer 2: Nice! 

@renatamottam should we add this section from now on?
